### PR TITLE
OSDOCS#4555: Add common terms for logging book

### PR DESCRIPTION
**Pull request by xenolinux**
Version(s): 4.8+

Issue: https://issues.redhat.com/browse/OSDOCS-4555

Link to docs preview: https://52781--docspreview.netlify.app/openshift-enterprise/latest/logging/cluster-logging.html#openshift-logging-common-terms_cluster-logging

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

**Comment by IshwarKanse** [comment]
@xenolinux You can refer the below logging related terms and description.
`annotation::
Annotations are used to attach metadata to objects.

Cluster Logging Operator::
The CLO (Cluster Logging Operator) provides a set of APIs to control collection and forwarding of logs from all pods and nodes in a cluster. This includes application logs (from regular pods), infrastructure logs (from system pods and node logs), and audit logs (special node logs with legal/security implications)

Custom Resource (CR)::
A CR is extension of the Kubernetes API. To configure OpenShift Logging and Log forwarding, you customize the clusterlogging and clusterlogforwader custom resource.

Event Router:: The Event Router is a pod that watches {product-title} events so they can be collected by OpenShift Logging.

Fluentd::
Fluentd is a log collector that resides on each OpenShift node to gather application, infrastructure and audit logs and forwards them to different outputs.

garbage collection::
The process of cleaning up cluster resources, such as terminated containers and images that are not referenced by any running pods.

Elasticsearch::
Elasticsearch is a distributed search and analytics engine, it  is used as the default log store for OpenShift Logging. 

Elasticsearch Operator::
Elasticsearch operator is used to run Elasticsearch cluster on top of OpenShift. The operator provide self-service for the Elasticsearch cluster operations an
d is used by OpenShift Logging.

indexing::
It is a data structure technique which is used to quickly locate and access data. Indexing optimizes the performance by minimizing the number of disk access required when a query is processed. 

JSON logging::
OpenShift Logging’s Log Forwarding API enables you to parse JSON logs into a structured object and forward them to either OpenShift Logging-managed Elasticsearch or any other third-party system supported by the Log Forwarding API.

Kibana:: Kibana is a browser-based console interface to query, discover, and visualize your Elasticsearch data through histograms, line graphs, pie charts, and other visualizations.

Kubernetes apiserver::
Kubernetes apiserver Validates and configures data for the API objects.

Labels::
You can use labels, which are key-value pairs, to organise and select subsets of objects, such as a pod.

Logging::
With OpenShift Logging you can aggregate application, infrastructure and audit logs throughout your cluster, store them to a default log store, forward them to third party systems and query, visualize the stored logs in the default log store. 

logging collector::
The Logging Collector collects logs from the cluster, formats them, and forwards them to the log store or third party systems. 

log store::
The log store is used to store aggregated logs. You can use the default Elasticsearch log store or forward logs to external log stores. The default log store is optimized and tested for short-term storage.

log visualizer::
This is the UI component you can use to view logs, graphs, charts, and so forth. The current implementation is Kibana. 

node::
A worker machine in the {product-title} cluster. A node is either a virtual machine (VM) or a physical machine depending on the cluster. Nodes are managed by control plane and contains services necessary to run pods. 

operators::
The preferred method of packaging, deploying, and managing a Kubernetes application in an {product-title} cluster. An Operator takes human operational knowledge and encodes it into software that is packaged and shared with customers.

pod::
One or more containers with shared resources, such as volume and IP addresses, running in your {product-title} cluster.
A pod is the smallest compute unit defined, deployed, and managed.

Role-based access control (RBAC)::
A key security control to ensure that cluster users and workloads have only access to resources required to execute their roles.

shards:: Elasticsearch organizes the log data from Fluentd into datastores, or indices, then subdivides each index into multiple pieces called shards.

taint::
Taints and tolerations ensure that pods are scheduled onto appropriate nodes. You can apply one or more taints on a node.

toleration::
You can apply tolerations to pods. Tolerations allow the scheduler to schedule pods with matching taints.

web console::
A user interface (UI) to manage {product-title}.
`

**Comment by jboxman-rh** [comment]
@xenolinux, some of this content overlaps with https://github.com/openshift/openshift-docs/pull/52812 and https://github.com/openshift/openshift-docs/pull/52810 and I wonder if using `snippets` for duplicate terms might reduce some of the duplication and make future maintenance easier?

File: modules/logging-common-terms.adoc [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+A logging collector collects logs from the cluster, formats them, and forwards them to the log store or third party systems.
+
+log store::
+A log store is used to store aggregated logs. You can use the default Elasticsearch log store or forward logs to external log stores. The default log store is optimized and tested for short-term storage.
+
+log visualizer::
+Log visualizer is the user interface (UI) component you can use to view logs, graphs, charts, and so forth. The current implementation is Kibana.
+
+node::
+A node is a worker machine in the {product-title} cluster. A node is either a virtual machine (VM) or a physical machine.node selector
```
Comment (jboxman-rh) [inline modules/logging-common-terms.adoc:63]: This sentence looks incomplete?

File: modules/logging-common-terms.adoc [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+Operators::
+Operators are the preferred method of packaging, deploying, and managing a Kubernetes application in an {product-title} cluster. An Operator takes human operational knowledge and encodes it into software that is packaged and shared with customers.
+
+pod::
+A pod is the smallest logical unit in Kubernetes. A pod is comprised of one or more containers to run in a worker node..
+
+Role-based access control (RBAC)::
+RBAC is a key security control to ensure that cluster users and workloads have only access to resources required to execute their roles.
```
Comment (jboxman-rh) [inline modules/logging-common-terms.adoc:72]:
```suggestion
RBAC is a key security control to ensure that cluster users and workloads have access only to resources required to execute their roles.
```

File: modules/logging-common-terms.adoc [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+shards::
+Elasticsearch organizes the log data from Fluentd into datastores, or indices, then subdivides each index into multiple pieces called shards.
+
+taint::
+Taints and tolerations ensure that pods are scheduled onto appropriate nodes. You can apply one or more taints on a node.
```
Comment (jboxman-rh) [inline modules/logging-common-terms.adoc:78]: This mentions _tolerations_ but that isn't described until later. Maybe omit _tolerations_ entirely here?

**Review by jboxman-rh — COMMENTED** [review]
@xenolinux, I've left some comments for your review. Thanks!

File: modules/logging-common-terms.adoc (hunk 3 above)
Comment (xenolinux) [reply modules/logging-common-terms.adoc]: acked

File: modules/logging-common-terms.adoc (hunk 2 above)
Comment (xenolinux) [reply modules/logging-common-terms.adoc]: acked

File: modules/logging-common-terms.adoc [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+Log visualizer is the user interface (UI) component you can use to view logs, graphs, charts, and so forth. The current implementation is Kibana.
+
+node::
+A node is a worker machine in the {product-title} cluster. A node is either a virtual machine (VM) or a physical machine.node selector
+
+Operators::
+Operators are the preferred method of packaging, deploying, and managing a Kubernetes application in an {product-title} cluster. An Operator takes human operational knowledge and encodes it into software that is packaged and shared with customers.
+
+pod::
+A pod is the smallest logical unit in Kubernetes. A pod is comprised of one or more containers to run in a worker node..
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply modules/logging-common-terms.adoc]: acked

File: modules/logging-common-terms.adoc (hunk 1 above)
Comment (xenolinux) [reply modules/logging-common-terms.adoc]: acked

File: modules/logging-common-terms.adoc [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+Kibana is a browser-based console interface to query, discover, and visualize your Elasticsearch data through histograms, line graphs, and pie charts.
+
+Kubernetes apiserver::
+Kubernetes apiserver Validates and configures data for the API objects.
+
+Labels::
+Labels are key-value pairs that you can use to organize and select subsets of objects such as a pod.
+
+Logging::
+With {product-title} Logging you can aggregate application, infrastructure and audit logs throughout your cluster, store them to a default log store, forward them to third party systems and query, visualize the stored logs in the default log store.
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply modules/logging-common-terms.adoc]: acked

[the remark this answers is not in the document]
Comment (xenolinux) [reply modules/logging-common-terms.adoc]: acked

File: modules/logging-common-terms.adoc [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+Indexing is a data structure technique which is used to quickly locate and access data. Indexing optimizes the performance by minimizing the amount of disk access required when a query is processed.
+
+JSON logging::
+{product-title} Logging Log Forwarding API enables you to parse JSON logs into a structured object and forward them to either {product-title} Logging-managed Elasticsearch or any other third-party system supported by the Log Forwarding API.
+
+Kibana::
+Kibana is a browser-based console interface to query, discover, and visualize your Elasticsearch data through histograms, line graphs, and pie charts.
+
+Kubernetes apiserver::
+Kubernetes apiserver Validates and configures data for the API objects.
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply modules/logging-common-terms.adoc]: acked

File: modules/logging-common-terms.adoc [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+indexing::
+Indexing is a data structure technique which is used to quickly locate and access data. Indexing optimizes the performance by minimizing the amount of disk access required when a query is processed.
+
+JSON logging::
+{product-title} Logging Log Forwarding API enables you to parse JSON logs into a structured object and forward them to either {product-title} Logging-managed Elasticsearch or any other third-party system supported by the Log Forwarding API.
+
+Kibana::
+Kibana is a browser-based console interface to query, discover, and visualize your Elasticsearch data through histograms, line graphs, and pie charts.
+
+Kubernetes apiserver::
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply modules/logging-common-terms.adoc]: acked

File: modules/logging-common-terms.adoc [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+The Cluster Logging Operator provides a set of APIs to control the collection and forwarding of application, infrastructure, and audit logs.
+
+Custom Resource (CR)::
+A CR is extension of the Kubernetes API. To configure {product-title} Logging and log forwarding, you can customize the `ClusterLogging` and the `ClusterLogForwarder` custom resource.
+
+event router::
+The event router is a pod that watches {product-title} events. It collects logs by using {product-title} Logging.
+
+Fluentd::
+Fluentd is a log collector that resides on each {product-title} node. It gathers application, infrastructure and audit logs and forwards them to different outputs.
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply modules/logging-common-terms.adoc]: acked

File: modules/logging-common-terms.adoc [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+This glossary defines common terms that are used in the architecture content.
+
+annotation::
+You can use annotations to attach metadata to objects.
+
+Cluster Logging Operator (CLO)::
+The Cluster Logging Operator provides a set of APIs to control the collection and forwarding of application, infrastructure, and audit logs.
+
+Custom Resource (CR)::
+A CR is extension of the Kubernetes API. To configure {product-title} Logging and log forwarding, you can customize the `ClusterLogging` and the `ClusterLogForwarder` custom resource.
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply modules/logging-common-terms.adoc]: acked

**Comment by xenolinux** [comment]
> @xenolinux, some of this content overlaps with #52812 and #52810 and I wonder if using `snippets` for duplicate terms might reduce some of the duplication and make future maintenance easier?

Hi @jboxman-rh sorry, I am missing something. Can you please clarify how can I use `snippets`?

**Comment by michaelryanpeter** [comment]
@xenolinux, you can find information about text snippets in the [Modular documentation reference guide](https://redhat-documentation.github.io/modular-docs/#using-text-snippets) and the [OSDOCS repo contributor guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#writing-text-snippets).

File: modules/logging-common-terms.adoc [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+With {product-title} Logging you can aggregate application, infrastructure, and audit logs throughout your cluster. You can also store them to a default log store, forward them to third party systems, and query and visualize the stored logs in the default log store.
+
+logging collector::
+A logging collector collects logs from the cluster, formats them, and forwards them to the log store or third party systems.
+
+log store::
+A log store is used to store aggregated logs. You can use the default Elasticsearch log store or forward logs to external log stores. The default log store is optimized and tested for short-term storage.
+
+log visualizer::
+Log visualizer is the user interface (UI) component you can use to view logs, graphs, charts, and so forth. The current implementation is Kibana.
```
Comment (michaelryanpeter) [inline modules/logging-common-terms.adoc:60]: See the ISG guidance on _[and so on](https://www.ibm.com/docs/en/ibm-style?topic=word-usage#and-so-on)_. WDYT of the following:

```suggestion
Log visualizer is the user interface (UI) component you can use to view information such as logs, graphs, charts, and other metrics. The current implementation is Kibana.
```

File: modules/logging-common-terms.adoc [hunk 11]
```diff
@@ -0,0 +1,83 @@
+// Module included in the following assemblies:
+//
+// * logging/cluster-logging.adoc
+
+:_content-type: REFERENCE
+[id="openshift-logging-common-terms_{context}"]
+= Common terms
+
+This glossary defines common terms that are used in the architecture content.
```
[the remark this answers is not in the document]
Comment (michaelryanpeter) [reply modules/logging-common-terms.adoc:9]: Is _logging subsystem_ the common term for these docs? It seems to vary between _OpenShift Containter Platform Logging_ and _logging subsystem_. I am not familiar with this area of the docs to know which term is the best to use for consistency.

**Review by michaelryanpeter — COMMENTED** [review]
I found a few small grammar and style guide issues. I think this PR would benefit from these changes, but they are not blockers. Feel free to ping me on Slack if you have questions.

/remove-label merge-review-in-progress
/remove-label merge-review-needed

File: modules/logging-common-terms.adoc (hunk 10 above)
Comment (xenolinux) [reply modules/logging-common-terms.adoc]: acked

File: modules/logging-common-terms.adoc [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+{product-title} Logging Log Forwarding API enables you to parse JSON logs into a structured object and forward them to either {product-title} Logging-managed Elasticsearch or any other third-party system supported by the Log Forwarding API.
+
+Kibana::
+Kibana is a browser-based console interface to query, discover, and visualize your Elasticsearch data through histograms, line graphs, and pie charts.
+
+Kubernetes API server::
+Kubernetes apiserver validates and configures data for the API objects.
+
+Labels::
+Labels are key-value pairs that you can use to organize and select subsets of objects such as a pod.
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply modules/logging-common-terms.adoc]: acked

File: modules/logging-common-terms.adoc [hunk 13]
```diff
@@ -0,0 +1,83 @@
...
+Indexing is a data structure technique which is used to quickly locate and access data. Indexing optimizes the performance by minimizing the amount of disk access required when a query is processed.
+
+JSON logging::
+{product-title} Logging Log Forwarding API enables you to parse JSON logs into a structured object and forward them to either {product-title} Logging-managed Elasticsearch or any other third-party system supported by the Log Forwarding API.
+
+Kibana::
+Kibana is a browser-based console interface to query, discover, and visualize your Elasticsearch data through histograms, line graphs, and pie charts.
+
+Kubernetes API server::
+Kubernetes apiserver validates and configures data for the API objects.
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply modules/logging-common-terms.adoc]: acked

File: modules/logging-common-terms.adoc [hunk 14]
```diff
@@ -0,0 +1,83 @@
...
+Garbage collection is the process of cleaning up cluster resources, such as terminated containers and images that are not referenced by any running pods.
+
+Elasticsearch::
+Elasticsearch is a distributed search and analytics engine. {product-title} uses ELasticsearch as a default log store for {product-title} Logging.
+
+Elasticsearch Operator::
+Elasticsearch operator is used to run Elasticsearch cluster on top of {product-title}. The Elasticsearch Operator provides self-service for the Elasticsearch cluster operations and is used by {product-title} Logging.
+
+indexing::
+Indexing is a data structure technique which is used to quickly locate and access data. Indexing optimizes the performance by minimizing the amount of disk access required when a query is processed.
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply modules/logging-common-terms.adoc]: acked

File: modules/logging-common-terms.adoc (hunk 11 above)
[the remark this answers is not in the document]
Comment (xenolinux) [reply modules/logging-common-terms.adoc]: Added OpenShift Containter Platform Logging, which is imo would be best to use for consistency

File: modules/logging-common-terms.adoc [hunk 15]
```diff
@@ -0,0 +1,83 @@
+// Module included in the following assemblies:
+//
+// * logging/cluster-logging.adoc
+
+:_content-type: REFERENCE
+[id="openshift-logging-common-terms_{context}"]
+= Common terms
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply modules/logging-common-terms.adoc]: Acked

[the remark this answers is not in the document]
Comment (xenolinux) [reply modules/logging-common-terms.adoc]: `Common {product-title} Logging terms` sounds good

**Pull request merged**